### PR TITLE
feat(iot-dev): Add configurable message expiration check job

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Previously, the client would only check for expired messages each time it tried to send a message. Now it will check for expired messages even if the client is idle.

### CODE REVIEW

File: iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java [hunk 1]
```diff
@@ -1766,20 +1816,8 @@ private void updateStatus(IotHubConnectionStatus newConnectionStatus, IotHubConn
             invokeConnectionStatusChangeCallback(newConnectionStatus, previousStatus, reason, throwable, deviceId);
         }
 
-        if (newConnectionStatus == IotHubConnectionStatus.CONNECTED)
-        {
-            try
-            {
-                correlationCallbackCleanupThread.start();
```
Comment (timtay-microsoft) [inline iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java:1773]: We shouldn't start/stop this thread when one device in a multiplexed connection is lost. Only start/stop it when the whole connection is opened/closed

File: iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java [hunk 2]
```diff
@@ -1913,28 +1951,6 @@ private boolean hasOperationTimedOut(long startTime)
         return (System.currentTimeMillis() - startTime) > this.getDefaultConfig().getOperationTimeout();
     }
 
-    /**
-     * Returns if the provided packet has lasted longer than the device operation timeout
-     *
-     * @return true if the packet has been in the queues for longer than the device operation timeout and false otherwise
-     */
-    private boolean hasOperationTimedOut(long startTime, String deviceId)
```
Comment (timtay-microsoft) [inline iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java:1921]: Unused method

**Review by patilsnr — APPROVED** [review]

**Pull request merged**